BUILD.bazel

```diff
@@ -1,34 +1,104 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+COMMON_FLAGS = [
+    "--std=c++11",
+    "-fPIC",
+    "-Wall",
+    "-Wextra",
+    "-Werror=return-type",
+    "-Wundef",
+    "-g",
+    "-O0",
+    "-gdwarf-4",
+    "-fPIC",
+]
+
+GCC_FLAGS = COMMON_FLAGS + [
+    "-Wuseless-cast",
+    "-Wmaybe-uninitialized",
+    "-fdiagnostics-color=always",
+]
+
+CLANG_FLAGS = COMMON_FLAGS + ["-fcolor-diagnostics"]
+
 cc_library(
     name = "cpptrace",
     srcs = glob([
         "src/**/*.hpp",
         "src/**/*.cpp",
     ]),
-    local_defines = [
-        "CPPTRACE_GET_SYMBOLS_WITH_LIBDWARF",
-        "CPPTRACE_DEMANGLE_WITH_CXXABI",
-        "CPPTRACE_UNWIND_WITH_LIBUNWIND"
-    ],
     hdrs = glob([
         "include/cpptrace/*.hpp",
         "include/ctrace/*.h",
     ]),
+    copts = select(
+        {
+            "@rules_cc//cc/compiler:msvc-cl": [
+                "/W4",
+                "/permissive-",
+                "/FS",
+            ],
+            "@rules_cc//cc/compiler:gcc": [
+                "-Wall",
+                "-Wextra",
+                "-Werror=return-type",
+                "-Wundef",
+                "-Wuninitialized",
+                "-fPIC",
+                "-std=c++11",
+            ],
+            "@rules_cc//cc/compiler:clang": CLANG_FLAGS,
+            "//conditions:default": COMMON_FLAGS,
+        },
+    ),
+    defines = ["CPPTRACE_STATIC_DEFINE"],
```

Comment:
This is an option that needs to be specified by anyone using the static cpptrace build, it shouldn't be set by cpptrace itself for cpptrace's sources.